lib/phlex/sgml.rb

```diff
@@ -454,6 +447,50 @@ def __attributes__(attributes, buffer = +"")
 			buffer
 		end
 
+		# @api private
+		#
+		# Provides the nested-attributes case for serializing out attributes.
+		# This allows us to skip many of the checks the `__attributes__` method must perform.
+		def __nested_attributes__(attributes, base_name, buffer = +"")
+			attributes.each do |k, v|
+				next unless v
+
+				name = case k
+											when String then k
+											when Symbol then k.name.tr("_", "-")
+											else raise ArgumentError.new("Attribute keys should be Strings or Symbols")
+				end
+
+				case v
+				when true
+					buffer << " " << base_name << name
+				when String
+					buffer << " " << base_name << name << '="' << v.gsub('"', "&quot;") << '"'
+				when Symbol
+					buffer << " " << base_name << name << '="' << v.name.gsub('"', "&quot;") << '"'
+				when Integer, Float
+					buffer << " " << base_name << name << '="' << v.to_s << '"'
+				when Hash
+					__nested_attributes__(v, "#{base_name}-#{name}-", buffer)
+				when Array
+					buffer << " " << base_name << name << '="' << v.compact.join(" ").gsub('"', "&quot;") << '"'
```

Comment:
I think we can use `gsub!` here to avoid a string allocation. Unfortunately that method doesn't return itself so it'll make the code a little uglier. Might be worth it though.

Reply:
Should I make that change in the existing `__attributes__` method as well?